vignettes/projecting_incidence.Rmd

```diff
@@ -0,0 +1,265 @@
+  dplyr::ungroup()
+```
+
+### Preparing the inputs  
+
+```{r linelist_gen, message=FALSE}
+days_since_index <- as.integer(covid19_sa$date - min(covid19_sa$date))
+
+start_times <- unlist(mapply(
+  function(x, y) rep(x, times = ifelse(y == 0, 1, y)),
+  days_since_index,
+  covid19_sa$cases
+))
+                       
+```
+
+
+Additionally, `chain_sim()` requires other inputs, which we will specify below: 
+
+```{r input_prep2, message=FALSE}
+#' Date to end simulation (14 day projection in this case)
+projection_window <- 14 # 14 days/ 2-week ahead projection
+
+projection_end_day <- max(days_since_index) + projection_window
+
+#' Number of simulations
+sim_rep <- 100
+
+#' Maximum chain size allowed
+chain_threshold <- 1000
+
+```
+
+#### Serial interval
+
+We also assume based on COVID-19 literature that the 
+serial interval, $si$, is lognormal distributed as follows:
+
+$ E[\text{si}] = \ln \left( \dfrac{\mu^2}{(\sqrt{\mu^2 + \sigma^2}} \right)$
+
+$ SD [\text{si}] = \sqrt {\ln \left(1 + \dfrac{\sigma^2}{\mu^2} \right)}$
+
+with $\mu = 4.7$ and standard deviation $\sigma = 2.9$.
+
+```{r input_prep3, message=FALSE}
+mu <- 4.7
+sigma <- 2.9
+
+si_sd <- sqrt(log(1 + (sigma / mu)^2)) # log standard deviation
+si_mean <- log((mu^2) / (sqrt(sigma^2 + mu^2))) # log mean
+
+#' serial interval function
+serial_interval <- function(sample_size) {
+  si <- rlnorm(sample_size, meanlog = si_mean, sdlog = si_sd)
+  return(si)
+}
+```
+
+#### Offspring distribution
+
+We assume an offspring distribution that is distributed as a negative binomial with $R = 2.5$ [@abbott2020] and $k = 0.58$ [@Wang2020]. In this parameterization, R represents the $\mathcal{R_0}$, which is defined as the average number of cases produced by a single individual in an entirely susceptible population. The parameter $k$ represents superspreading, that is, the degree of heterogeneity in transmission by single individuals.
+
+### Simulations
+To summarize the simulation set up, for each of the `r sim_rep` simulations, we want to project cases over a `r projection_window` day period since the last case, assuming that no chain would exceed `r chain_threshold`. 
+
+#### Model assumptions
+
+`chain_sim()` makes the following simplifying assumptions:
+
+1. All cases are observed
+1. There is no reporting delay
+1. Reporting rate is constant through the course of the epidemic
+1. No interventions have been implemented
+1. Population is homogeneous and well-mixed
+
+```{r simulations, message=FALSE}
+set.seed(1234)
+
+
+sim_chain_sizes <- lapply(
+  seq_len(sim_rep),
+  function(sim) {
+    chain_sim(
+      n = length(start_times),
+      offspring = "nbinom",
+      mu = 2.5,
+      size = 0.58,
+      stat = "size",
+      infinite = chain_threshold,
+      serial = serial_interval,
+      t0 = start_times,
+      tf = projection_end_day,
+      tree = TRUE
+    ) %>%
+      mutate(sim = sim)
+  }
+)
+
+sim_output <- do.call(rbind, sim_chain_sizes)
```

Comment:
Since you are using {dplyr}, could also use `bind_rows()` here for better readability.

Reply:
True.